### PR TITLE
 collapse into consistent operator client interfaces

### PR DESCRIPTION
**Pull request by deads2k**
This lets us create one client interface with non-overlapping methods to support multiple helper functions.

/assign @sttts

### CODE REVIEW

File: pkg/operator/resourcesynccontroller/resourcesync_controller.go [hunk 1]
```diff
@@ -5,6 +5,8 @@ import (
 	"sync"
 	"time"
 
+	"github.com/openshift/library-go/pkg/operator/v1helpers"
```
Comment (sttts) [inline pkg/operator/resourcesynccontroller/resourcesync_controller.go]: order

Comment (sttts) [reply pkg/operator/resourcesynccontroller/resourcesync_controller.go]: also below in the other files

File: pkg/operator/v1helpers/interfaces.go [hunk 2]
```diff
@@ -2,10 +2,18 @@ package v1helpers
...
 type OperatorClient interface {
+	Informer() cache.SharedIndexInformer
 	GetOperatorState() (spec *operatorv1.OperatorSpec, status *operatorv1.OperatorStatus, resourceVersion string, err error)
 	UpdateOperatorSpec(string, *operatorv1.OperatorSpec) (spec *operatorv1.OperatorSpec, resourceVersion string, err error)
 	UpdateOperatorStatus(string, *operatorv1.OperatorStatus) (status *operatorv1.OperatorStatus, resourceVersion string, err error)
 }
+
+type StaticPodOperatorClient interface {
+	OperatorClient
+	GetStaticPodOperatorState() (*operatorv1.OperatorSpec, *operatorv1.StaticPodOperatorStatus, string, error)
```
Comment (sttts) [inline pkg/operator/v1helpers/interfaces.go]: s/State/Status/

File: pkg/operator/v1helpers/interfaces.go [hunk 3]
```diff
@@ -2,10 +2,18 @@ package v1helpers
 
 import (
 	operatorv1 "github.com/openshift/api/operator/v1"
+	"k8s.io/client-go/tools/cache"
 )
 
 type OperatorClient interface {
+	Informer() cache.SharedIndexInformer
 	GetOperatorState() (spec *operatorv1.OperatorSpec, status *operatorv1.OperatorStatus, resourceVersion string, err error)
```
Comment (sttts) [inline pkg/operator/v1helpers/interfaces.go]: s/State/Status/

File: pkg/operator/v1helpers/helpers.go [hunk 4]
```diff
@@ -2,8 +2,11 @@ package v1helpers
 
 import (
 	"fmt"
+	"strings"
 	"time"
 
+	errors2 "k8s.io/apimachinery/pkg/util/errors"
```
Comment (sttts) [inline pkg/operator/v1helpers/helpers.go]: utilerrors

File: pkg/operator/v1helpers/test_helpers.go.go [hunk 5]
```diff
@@ -1,20 +1,19 @@
...
+package v1helpers
 
 import (
 	"fmt"
 	"strconv"
 	"time"
 
-	"k8s.io/api/core/v1"
+	"github.com/openshift/api/operator/v1"
+	v13 "k8s.io/api/core/v1"
```
Comment (sttts) [inline pkg/operator/v1helpers/test_helpers.go.go]: that's a nice alias, also v14 below

Comment (deads2k) [reply pkg/operator/v1helpers/test_helpers.go.go]: fun, auto refactor does weird stuff

File: pkg/operator/v1helpers/interfaces.go (hunk 2 above)
Comment (deads2k) [reply pkg/operator/v1helpers/interfaces.go]: > s/State/Status/

it's both spec and status, so state

File: pkg/operator/v1helpers/interfaces.go (hunk 3 above)
Comment (deads2k) [reply pkg/operator/v1helpers/interfaces.go]: > s/State/Status/

it's both spec and status, so state

Comment (sttts) [reply pkg/operator/v1helpers/interfaces.go]: misread.

**Pull request merged**